vm.go

```diff
@@ -701,6 +769,82 @@ func (vm *VirtualMachineSpec) String() string {
 	return string(jsonString)
 }
 
+func (vm *VirtualMachineSpec) modifySetting(ctx context.Context, doc *hcsschema.ModifySettingRequest) (err error) {
+	if doc.GuestRequest == nil || vm.gc == nil {
+		return vm.system.Modify(ctx, doc)
+	}
+
+	hostdoc := *doc
+	hostdoc.GuestRequest = nil
+	if doc.ResourcePath != "" && doc.RequestType == requesttype.Add {
+		err = vm.system.Modify(ctx, &hostdoc)
+		if err != nil {
+			return fmt.Errorf("adding VM resources: %s", err)
+		}
+		defer func() {
+			if err != nil {
+				hostdoc.RequestType = requesttype.Remove
+				rerr := vm.system.Modify(ctx, &hostdoc)
+				if rerr != nil {
+					log.G(ctx).WithError(err).Error("failed to roll back resource add")
+				}
+			}
+		}()
+	}
+	err = vm.gc.Modify(ctx, doc.GuestRequest)
+	if err != nil {
+		return fmt.Errorf("guest modify: %s", err)
+	}
+	if doc.ResourcePath != "" && doc.RequestType == requesttype.Remove {
+		err = vm.system.Modify(ctx, &hostdoc)
+		if err != nil {
+			err = fmt.Errorf("removing VM resources: %s", err)
+			log.G(ctx).WithError(err).Error("failed to remove host resources after successful guest request")
+			return err
+		}
+	}
+	return nil
+}
+
+func (vm *VirtualMachineSpec) listenVsock(port uint32) (net.Listener, error) {
+	return winio.ListenHvsock(&winio.HvsockAddr{
+		VMID:      vm.runtimeId,
+		ServiceID: winio.VsockServiceID(port),
+	})
+}
+
+// acceptAndClose accepts a connection and then closes a listener. If the
+// context becomes done or the VM terminates, the operation will be
```

Comment:
nit: context becomes done -> does it mean "context completed or available"?

Reply:
context becomes done means "ctx timeout"

caller provides context, so context is always available, we don't do anything to modify context, so context won't completed in this function.